lib/operations/collection_ops.js

```diff
@@ -230,12 +230,9 @@ function countDocuments(coll, query, options, callback) {
   delete options.limit;
   delete options.skip;
 
-  coll.aggregate(pipeline, options, (err, result) => {
+  coll.aggregate(pipeline, options).toArray((err, docs) => {
     if (err) return handleCallback(callback, err);
-    result.toArray((err, docs) => {
-      if (err) return handleCallback(err);
-      handleCallback(callback, null, docs.length ? docs[0].n : 0);
-    });
+    handleCallback(callback, null, docs.length ? docs[0].n : 0);
```

Comment:
If `docs` is an empty array there will be an error

Reply:
@EdikDolynskyi I don't think that is the case here. if docs is an empty array, `docs.length` will be 0.